rules/linux/credential_access_unusual_instance_metadata_service_api_request.toml

```diff
@@ -42,14 +41,27 @@ sequence by host.id,  process.parent.entity_id with maxspan=1s
       "/etc/cron*", "/etc/update-motd.d/*", "/boot/*", "/srv/*", "/run/*", "/etc/rc.local"
     ) or
     process.command_line: "*169.254.169.254*" and
-    not process.working_directory: (
-      "/opt/rapid7*",
-      "/opt/nessus*",
-      "/snap/amazon-ssm-agent*",
-      "/srv/snp/docker/overlay2*",
-      "/var/log/amazon/ssm*"
-    )]
-[network where host.os.type == "linux" and event.action == "connection_attempted" and destination.ip == "169.254.169.254"]
+    not (process.working_directory: (
+          "/opt/rapid7*",
+          "/opt/nessus*",
+          "/snap/amazon-ssm-agent*",
+          "/var/snap/amazon-ssm-agent/*",
+          "/var/log/amazon/ssm/*",
+          "/srv/snp/docker/overlay2*",
+          "/opt/nessus_agent/var/nessus/*") or
+        process.executable: (
+          "/opt/rumble/bin/rumble-agent*",
+          "/opt/aws/inspector/bin/inspectorssmplugin") or
+        process.parent.executable: (
+          "/usr/bin/setup-policy-routes",
+          "/usr/share/ec2-instance-connect/*",
+          "/var/lib/amazon/ssm/*")
+        )
+]
+[network where host.os.type == "linux"
+  and event.action == "connection_attempted"
+  and destination.ip == "169.254.169.254"
+  and not process.command_line: "*intelephense/lib/intelephense.js*"]
```

Comment:
Is there any way to not prefix the command line with a `*`? So use the full command line or at least a part from the start to enhance performance there?

Reply:
This one was rather odd, it's the only instance of a VSCode extension making IMDS calls in the background. It is intellisense for PHP servers, where it hits IMDS regardless of CSP. I pulled the code and reviewed it, but hard to indicate why it is doing this especially from a developer's endpoint.

I'll check again on a better exclusion.

Reply:
Unfortunately only indication are from network alerts. The volume is relatively low and an edge-case. Rather than whitelisting, we can allow those with these instances to add custom exceptions if necessary.